packages/wagmi/src/index.ts

```diff
@@ -49,14 +92,10 @@ export function bindings(): IBindings {
         const { chain } = getNetwork();
 
         if (chain?.id !== chainId) {
-          try {
-            await switchNetwork({ chainId });
-          } catch (err) {
-            // best effort to switch network, if it fails, we just ignore it
-            if (!(err instanceof SwitchChainNotSupportedError)) {
-              throw err;
-            }
-          }
+          const walletClient = await getWalletClient();
+          invariant(walletClient, 'Wallet client not found');
+          await switchChain({ walletClient, chainId });
+          return signerFromWalletClient({ walletClient, chainId });
```

Comment:
There is logic already doing this in the `SignerFactory`, rather than replicating it a this level, can we see why it does not seem to reach the underlying `window.ethereum` provider?

Reply:
(through these bindings)

Reply:
thanks, much simplifier fix in the end to avoid the error being thrown in these bindings